source/plugin/blocks/accessing.rst

```diff
@@ -2,217 +2,150 @@
+    org.spongepowered.api.world.server.ServerWorld
+    org.spongepowered.api.state.BooleanStateProperty
+    org.spongepowered.api.data.DataManipulator.Immutable
 
 Basic Information
 ~~~~~~~~~~~~~~~~~
 
 Blocks are most commonly identified and accessed by their :javadoc:`Location`. This location points to a certain
-coordinate within an :javadoc:`Extent`. In most cases a :javadoc:`World` will be used as the ``Extent``.
+coordinate within an :javadoc:`World`. In most cases a :javadoc:`ServerWorld` will be used as the ``World``.
 
 .. code-block:: java
 
     import org.spongepowered.api.Sponge;
-    import org.spongepowered.api.world.Location;
-    import org.spongepowered.api.world.World;
+    import org.spongepowered.api.world.server.ServerLocation;
+    import org.spongepowered.api.world.server.ServerWorld;
+    import org.spongepowered.api.ResourceKey;
 
-    public Location<World> getBlockAt(String worldName, int posX, int posY, int posZ) {
-        World world = Sponge.getServer().getWorld(worldName).get();
-        Location<World> blockLoc = new Location<World>(world, posX, posY, posZ);
+    public ServerLocation getBlockAt(ResourceKey worldKey, int posX, int posY, int posZ) {
+        ServerWorld world = Sponge.server().worldManager().world(worldKey).get();
+        ServerLocation blockLocation = world.location(posX, posY, posZ);
         return blockLoc;
     }
 
 .. warning::
 
-    Note that the above example does not check if the world exists. ``getWorld(worldName).get()`` will fail if there
+    Note that the above example does not check if the world exists. ``world(worldKey).get()`` will fail if there
     is no world of that name loaded.
 
 
-With this ``Location`` object you can then obtain further information about the block. The following code checks if a
+With this ``ServerLocation`` object you can then obtain further information about the block. The following code checks if a
 referenced block is any kind of banner by checking the blocks type.
 
 .. code-block:: java
 
     import org.spongepowered.api.block.BlockType;
     import org.spongepowered.api.block.BlockTypes;
+    import org.spongepowered.api.tag.BlockTypeTag;
+    import org.spongepowered.api.world.server.ServerLocation;
 
-    public boolean isBanner(Location<World> blockLoc) {
-        BlockType type = blockLoc.getBlock().getType();
-        return type.equals(BlockTypes.STANDING_BANNER)
-                || type.equals(BlockTypes.WALL_BANNER);
+    public boolean isBanner(ServerLocation blockLoc) {
+        Tag<BlockType> bannerTag = BlockTypeTags.BANNERS.get();
+        BlockType blockType = blockLoc.blockType();
+        return bannerTag.contains(blockType);
     }
 
 .. tip::
     
-    The function ``==`` could be used in place of ``equals()`` as there is only one :javadoc:`BlockType` instance for
+    The function ``==`` could be used in place of ``equals()`` when it comes to comparing two :javadoc:`BlockType`'s as there are a fixed number of instance for
```

Comment:
instance -> instances. Also please try to keep line lengths shorter (> 120), it gets hard to proofread.

Reply:
thanks, changed